modules/seqexec/server/src/main/scala/seqexec/server/package.scala

```diff
@@ -37,32 +32,44 @@ import squants.Time
 
 package server {
   @Lenses
-  final case class EngineState(queues: ExecutionQueues, selected: Map[Instrument, Observation.Id], conditions: Conditions, operator: Option[Operator], sequences: Map[Observation.Id, SequenceData[IO]])
-
-  // TODO EngineState extending Engine.State is problematic when trying to remove the strong IO dependency
-  object EngineState extends Engine.State[IO, EngineState]{
-    val default: EngineState =
-      EngineState(
+  final case class EngineState[F[_]](
+    queues: ExecutionQueues,
+    selected: Map[Instrument, Observation.Id],
+    conditions: Conditions,
+    operator: Option[Operator],
+    sequences: Map[Observation.Id, SequenceData[F]]
+  )
+
+  object EngineState {
+    def default[F[_]]: EngineState[F] =
+      EngineState[F](
         Map(CalibrationQueueId -> ExecutionQueue.init(CalibrationQueueName)),
         Map.empty,
         Conditions.Default,
-        None, Map.empty)
+        None,
+        Map.empty
+      )
 
-    def instrumentLoadedL(
+    def instrumentLoadedL[F[_]](
       instrument: Instrument
-    ): Lens[EngineState, Option[Observation.Id]] =
-      GenLens[EngineState](_.selected) ^|-> at(instrument)
+    ): Lens[EngineState[F], Option[Observation.Id]] =
+      GenLens[EngineState[F]](_.selected) ^|-> at(instrument)
 
-    def atSequence(sid:Observation.Id): Optional[EngineState, SequenceData[IO]] =
+    def atSequence[F[_]](sid:Observation.Id): Optional[EngineState[F], SequenceData[F]] =
       EngineState.sequences ^|-? index(sid)
 
-    override def sequenceStateIndex(
-      sid: Observation.Id
-    ): Optional[EngineState, Sequence.State[IO]] =
-      atSequence(sid) ^|-> SequenceData.seq
+    def sequenceStateIndex[F[_]](sid: Observation.Id): Optional[EngineState[F], Sequence.State[F]] =
+      atSequence[F](sid) ^|-> SequenceData.seq
+
+    def engineState[F[_]]: Engine.State[F, EngineState[F]] = new Engine.State[F, EngineState[F]] {
```

Comment:
Would it be a problem that we produce a new engine state every time we call this?

Reply:
No really. It is the implementation of a `trait` with no data.